gp-includes/routes/translation.php

```diff
@@ -28,8 +28,15 @@ public function import_translations_get( $project_path, $locale_slug, $translati
 			return $this->die_with_404();
 		}
 
-		if ( $this->cannot_and_redirect( 'approve', 'translation-set', $translation_set->id ) ) {
-			return;
+		$can_import = $this->can( 'import', 'translation-set', $translation_set->id );
+
+		if ( ! $can_import ) {
+			$can_import_waiting = $this->can( 'import-waiting', 'translation-set', $translation_set->id );
+
+			if ( ! $can_import_waiting ) {
+				$this->redirect_with_error( __( 'You are not allowed to do that!', 'glotpress' ), $url );
+				return;
+			}
 		}
```

Comment:
By default, you should assume the user has no permissions and then add them to the user by granting them.  The above permissions, for example, don't allow you to grant someone only rights to import current (aka the current behaviour).